api/document/js/main.ts

```diff
@@ -0,0 +1,83 @@
+import {Schema, Node} from "prosemirror-model";
+import {EditorView} from "prosemirror-view";
+import {EditorState} from "prosemirror-state";
+import {menuBar, undoItem, redoItem, MenuItem} from "prosemirror-menu";
+import {history, undo, redo} from "prosemirror-history";
+import {keymap} from "prosemirror-keymap";
+import {baseKeymap} from "prosemirror-commands";
+
+declare function require(path: string): null;
+
+require("prosemirror-view/style/prosemirror.css");
+require("prosemirror-menu/style/menu.css");
+
+const EDITOR_ID = 'editor';
+
+const EXAMPLE_DOC = {
+  type: 'doc',
+  content: [{
+    type: 'paragraph',
+    content: [{
+      type: 'text',
+      text: 'Hello, I am a paragraph.',
+    }],
+  }],
+};
+
+const schema = new Schema({
+  nodes: {
+    doc: {
+      content: 'paragraph+',
+    },
+    paragraph: {
+      content: 'inline*',
+      toDOM() { return ['p', 0]; },
+    },
+    text: {
+      group: 'inline',
+    },
+  },
+  marks: {},
+});
+
+function getDocument(): Node {
+  const doc = Node.fromJSON(schema, EXAMPLE_DOC);
+
+  doc.check();
```

Comment:
Is this validating the document against that schema? If so, what happens when it fails? Is this a placeholder for something else?

Reply:
Yep, it's validating the document against the schema and raising an exception if it fails--it's really just there as a sanity check, though I suppose in production we might want to make it catch the exception and report an error or something, rather than crashing the whole front-end.

I'll add a comment to this effect.